src/tls_migration.erl

```diff
@@ -0,0 +1,67 @@
+%% TLS migration script (temporary)
+%%
+%% Run from the console as:
+%% > l(tls_migration). tls_migration:main(ok).
+
+-module(tls_migration).
+
+-export([main/1]).
+
+-include_lib("ex_uri/include/ex_uri.hrl").
+-include("logplex_drain.hrl").
+-include("logplex_logging.hrl").
+
+main(_) ->
+    ets:foldl(fun(Drain, _Acc) ->
+                      case should_migrate_drain(Drain) of
+                          true ->
+                              migrate_drain_maybe(Drain);
+                          _ ->
+                              ok
+                      end
+              end, notused, drains),
+    ok.
+
+migrate_drain_maybe(Drain) ->
+    case attempt_connection(Drain) of
+        {error, Reason} ->
+            handle_failed_drain(Drain, Reason);
+        _ ->
+            ok
+    end.
+
+handle_failed_drain(Drain, Reason) ->
+    write_csv_row(Drain, Reason),
+    toggle_insecure_and_save(Drain).
+
+attempt_connection(#drain{id=DrainID, channel_id=ChannelID, uri=#ex_uri{authority=#ex_uri_authority{host=Host, port=Port}} = URI}) ->
+    io:format("Attempting connection to ~p:~p for drain ~p~n", [Host, Port, DrainID]),
+    case logplex_tlssyslog_drain:do_connect(Host, Port, URI, DrainID, ChannelID) of
```

Comment:
We should test an #insecure connect first, to ensure the drain works in the first place. Followed by the newer set of options.